src/wagtail_vector_index/ai_utils/backends/llm.py

```diff
@@ -77,10 +77,23 @@ class LLMChatBackend(BaseChatBackend[LLMChatBackendConfig]):
     config: LLMChatBackendConfig
     config_cls = LLMChatBackendConfig
 
-    def chat(self, *, user_messages: Sequence[str]) -> AIResponse:
+    def chat(
+        self, *, messages: Sequence[ChatMessage], stream: bool = False, **kwargs
+    ) -> AIResponse:
+        if stream:
+            raise NotImplementedError(
+                "Streaming chat is not currently supported by this backend."
+            )
+
         model = self._get_llm_chat_model()
-        full_prompt = os.linesep.join(user_messages)
-        return model.prompt(full_prompt, **self._get_prompt_kwargs())
+        full_prompt = os.linesep.join([message["content"] for message in messages])
+        text_response = model.prompt(full_prompt, **self._get_prompt_kwargs()).text()
+        return AIResponse(choices=[text_response])
+
+    def achat(
+        self, *, messages: Sequence[ChatMessage], stream: bool = False, **kwargs
+    ) -> AIResponse:
+        raise NotImplementedError("Async chat is not supported by this backend.")
```

Comment:
Is there a reason we're not making this the default implementation (on `BaseChatBackend`)?

Reply:
No that makes sense!